src/pages/CoffeeEdit.js

```diff
@@ -1,8 +1,82 @@
-import React from 'react'
+import { useParams, useNavigate } from 'react-router-dom'
+import React, { useState} from 'react'
+import { FormGroup, Label, Input, Form, Button } from 'reactstrap'
 
-const CoffeeEdit = () => {
-  return (
-    <div>CoffeeEdit</div>
+
+const CoffeeEdit = ({ coffees, updateCoffee }) => {
+  const { id } = useParams()
+  let currentCoffee = coffees?.find((coffee) => coffee.id === +id)
+  
+  const navigate = useNavigate()
+
+  const [editCoffee, setEditCoffee] = useState({
+    name: currentCoffee.name,
+    size: currentCoffee.size,
+    bio: currentCoffee.bio,
+    image: currentCoffee.image,
+    milk: currentCoffee.milk
+  })
+
+  const handleChange = (e) => {
+    setEditCoffee({ ...editCoffee, [e.target.name]: e.target.value })
+  }
+
+const handleSubmit = () => {
+  updateCoffee(editCoffee, currentCoffee.id)
+  navigate("/coffeeindex")
+}
+return (
+  <>
+    <Form>
+        <FormGroup>
+          <Label for="name">
+            Name
+          </Label>
+          <Input
+            name="name"
+            onChange={handleChange}
+            placeholder="What is your name?"
```

Comment:
For update, oftentimes you will autofill the value in the input box.  This can be done with the value or default value attribute instead of the placeholder.  This helps the user know what they are updating.